app/components/markdown/markdown_link/markdown_link.js

```diff
@@ -55,7 +57,16 @@ export default class MarkdownLink extends PureComponent {
         const match = matchDeepLink(url, serverUrl, siteURL);
         if (match) {
             if (match.type === DeepLinkTypes.CHANNEL) {
-                this.props.actions.handleSelectChannelByName(match.channelName, match.teamName);
+                const response = await this.props.actions.handleSelectChannelByName(match.channelName, match.teamName);
+
+                if (response && response.error) {
```

Comment:
by making the return of the action consistent you can check for `error` without validating that the response is not null/undefined.